molo/yourwords/tests/test_views.py

```diff
@@ -1,93 +1,47 @@
-from django.contrib.auth.models import User
-from django.contrib.contenttypes.models import ContentType
 from django.test import TestCase
 from django.test.client import Client
 from django.core.urlresolvers import reverse
 
-from molo.core.models import LanguagePage, Main
+from molo.core.tests.base import MoloTestCaseMixin
+from molo.core.models import SiteLanguage
 from molo.yourwords.models import (
     YourWordsCompetition, YourWordsCompetitionEntry)
-from wagtail.wagtailcore.models import Site, Page
 
 
-class TestYourWordsViewsTestCase(TestCase):
+class TestYourWordsViewsTestCase(MoloTestCaseMixin, TestCase):
 
     def setUp(self):
-        self.user = User.objects.create_user(
-            username='tester',
-            email='tester@example.com',
-            password='tester')
-
-        # Create page content type
-        page_content_type, created = ContentType.objects.get_or_create(
-            model='page',
-            app_label='wagtailcore'
-        )
-
-        # Create root page
-        Page.objects.create(
-            title="Root",
-            slug='root',
-            content_type=page_content_type,
-            path='0001',
-            depth=1,
-            numchild=1,
-            url_path='/',
-        )
-
-        main_content_type, created = ContentType.objects.get_or_create(
-            model='main', app_label='core')
-
-        # Create a new homepage
-        main = Main.objects.create(
-            title="Main",
-            slug='main',
-            content_type=main_content_type,
-            path='00010001',
-            depth=2,
-            numchild=0,
-            url_path='/home/',
-        )
-        main.save_revision().publish()
-
-        self.english = LanguagePage(
-            title='English',
-            code='en',
-            slug='english')
-        main.add_child(instance=self.english)
-        self.english.save_revision().publish()
-
-        # Create a site with the new homepage set as the root
-        Site.objects.all().delete()
-        Site.objects.create(
-            hostname='localhost', root_page=main, is_default_site=True)
+        self.user = self.login()
+        self.mk_main()
+        # Creates Main language
+        self.english = SiteLanguage.objects.create(locale='en')
 
     def test_yourwords_competition_page(self):
         client = Client()
-        client.login(username='tester', password='tester')
+        client.login(username='superuser', password='pass')
 
         comp = YourWordsCompetition(
             title='Test Competition',
             description='This is the description',
             slug='test-competition')
-        self.english.add_child(instance=comp)
+        self.main.add_child(instance=comp)
         comp.save_revision().publish()
 
         comp = YourWordsCompetition.objects.get(slug='test-competition')
 
-        response = client.get('/english/test-competition/')
+        response = client.get('/test-competition/')
```

Comment:
Please also add a test here to check that submissions to a different language should be stored against the main language. e.g `/test-competition-fr/`